rewrite-java-lombok/src/main/java/org/openrewrite/java/lombok/LombokSupport.java

```diff
@@ -59,8 +59,13 @@ public class LombokSupport {
         }
 
         String oldValue = System.setProperty("shadow.override.lombok", String.join(File.pathSeparator, overrideClasspath));
+        Class<?> shadowLoaderClass;
+        try {
+            shadowLoaderClass = Class.forName("lombok.launch.ShadowClassLoader", true, parserClassLoader);
+        } catch (Exception e) {
+            shadowLoaderClass = Class.forName("lombok.launch.ShadowClassLoader", true, parserClassLoader.getParent());
+        }
         try {
```

Comment:
Why pull this out of the original `try`? Now it looks like it could happen that the finally below is not called to restore/clear the property.

Reply:
On top of that, please add a comment why the fallback is there in the first place. Something like:
```java
// `rewrite-java-lombok` is shaded within the cli's `modmaven` module, thus is only visible in the parent class loader
```
will do.

Reply:
I realized yesterday the way I changed things isn't valid, I'll move trying to create the ShadowClassLoader back into the original try

I don't think I want to expose the CLI context in OpenRewrite, but I'll try to add a comment without that